### PR TITLE
More accessibility checks

### PR DESCRIPTION
**Pull request by dfaulken**
Even more checks will cause builds to fail if they don't pass. The only ones that I needed to fix were contrast issues, again with Rails-native stuff.

My suspicion is that some checks aren't being triggered because Capybara isn't interacting with potentially offending elements. Perhaps it might be nice to make more assertions about what kind of content should be on each page, thereby causing those checks to run on those selectors.

### CODE REVIEW

File: app/assets/stylesheets/layout.scss [hunk 1]
```diff
@@ -62,6 +62,14 @@ body{
     background: $main-panel-background;
     margin: 1em;
 
+    a:hover{
```
Comment (dfaulken) [inline app/assets/stylesheets/layout.scss]: Previously it was black for some reason, which, yeah, is nearly illegible with dark blue text.

File: app/assets/stylesheets/layout.scss [hunk 2]
```diff
@@ -62,6 +62,14 @@ body{
...
+      background-color: inherit;
+    }
+
+    .field_with_errors{
```
Comment (dfaulken) [inline app/assets/stylesheets/layout.scss]: Rails-native, it's just `red;` e.g. `ff0000`, which isn't a good contrast ratio with black text (5.3:1).

File: app/assets/stylesheets/layout.scss [hunk 3]
```diff
@@ -71,7 +79,7 @@ body{
       @include default-borders(1px);
 
       &.delete{
-        background: #cdacac;
+        background: #e4d2d2;
```
Comment (dfaulken) [inline app/assets/stylesheets/layout.scss]: Text color here is `#333`, so the original background color had a ratio of 6.1:1 (AA but not AAA). This is 8.7:1.

File: app/views/users/index.haml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 - if @inactive
-  %h1 Manage Inactive Drivers
+  %h1 Inactive Drivers
```
Comment (dfaulken) [inline app/views/users/index.haml]: Not for accessibility, I just thought it was confusing to have an h1 saying 'Manage Inactive Drivers' immediately followed by a link reading 'Manage Active Drivers', for example. So I took the imperative verb out of the title.

File: test/application_system_test_case.rb [hunk 5]
```diff
@@ -4,6 +4,20 @@
...
+  'AX_HTML_02',  # An element's ID must be unique in the DOM
+  'AX_TEXT_01',  # Controls and media elements should have labels
+  'AX_TEXT_02',  # Images should have an alt attribute, unless they have an ARIA
+                 # role of 'presentation'
+  'AX_FOCUS_03', # Avoid positive integer values for tabindex
+  'AX_COLOR_01', # Text elements should have a reasonable contrast ratio
+  'AX_TABLE_01', # Tables must have appropriate headers
+]
+# rubocop:enable Layout/CommentIndentation
+
```
Comment (dfaulken) [inline test/application_system_test_case.rb]: There are [several other checks](https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules) — particularly relating to `aria-` attributes (which I don't even use). For many of them though, the description just reads 'stub', so I'm unclear whether they're actually implemented in the JS library.

File: app/assets/stylesheets/layout.scss [hunk 6]
```diff
@@ -62,6 +62,14 @@ body{
     background: $main-panel-background;
     margin: 1em;
 
+    a:hover{
+      background-color: inherit;
+    }
+
+    .field_with_errors{
+      background: #ff9999;
```
Comment (dfaulken) [inline app/assets/stylesheets/layout.scss]: Rails-native, it's just `red;` e.g. `#f00`, which isn't a good contrast ratio with black text (5.3:1, so passes AA but fails AAA). This is 10.3:1, which passes WCAG 2.0 AAA as well. 

Do we think I should write this as `#f99`, btw?

File: app/assets/stylesheets/incidents.scss [hunk 7]
```diff
@@ -71,3 +71,8 @@ table.history>tbody>tr{
     }
   }
 }
+
+.high-contrast-options option{
+  color: black;
+  background: white;
```
Comment (dfaulken) [inline app/assets/stylesheets/incidents.scss]: I'm a little confused about option styling in general. I thought how those appear is pretty browser-dependent.

File: app/assets/stylesheets/layout.scss (hunk 6 above)
Comment (Anbranin) [inline app/assets/stylesheets/layout.scss]: You _can_ but it's fine to not change it.

File: app/assets/stylesheets/incidents.scss (hunk 7 above)
Comment (Anbranin) [inline app/assets/stylesheets/incidents.scss]: Maybe it's browser-dependent unless it's specified, so it's safer to do so?

Comment (werebus) [inline app/assets/stylesheets/incidents.scss]: Yeah, it's a bit of a crap-shoot. Styling form elements has always been terrible which is why so many UI frameworks would just replace them with lists and divs and such.

Interestingly, on *my* version of Firefox, **setting** a background drastically changes how the element looks:

screenshot_2017-06-12_12-59-51
screenshot_2017-06-12_13-01-14

Dunno.

**Pull request merged**